nixos/modules/system/etc/setup-etc.pl

```diff
@@ -12,7 +13,13 @@ sub atomicSymlink {
     my ($source, $target) = @_;
     my $tmp = "$target.tmp";
     unlink $tmp;
-    symlink $source, $tmp or return 0;
+    # Create relative symlinks, so that the links can be followed if
+    # the NixOS installation is not mounted as filesystem root.
+    # Absolute symlinks violate the os-release format
+    # at https://www.freedesktop.org/software/systemd/man/os-release.html
+    # and break e.g. systemd-nspawn and os-prober.
+    my $rel = File::Spec->abs2rel($source, dirname $target);
+    symlink $rel, $tmp or return 0;
```

Comment:
This is wrong. This works for system
```
$ ls -ld /etc/static
lrwxrwxrwx 1 root root 53 січ 23 21:27 /etc/static -> ../nix/store/23r7mdcbqh5h7rfvdzifcygjfp9f5kyd-etc/etc
```
But fails for container:
```
# ls -ld /var/lib/containers/db2/etc/static
lrwxrwxrwx 1 root root   53 Jan 23 15:07 /var/lib/containers/db2/etc/static -> ../nix/store/c7xn0a16907d2mi3kyidina9hg3r9nh4-etc/etc
```
That symlink doesn't resolve for container, when viewed from host